core/src/main/java/org/apache/calcite/rel/core/Project.java

```diff
@@ -145,6 +146,12 @@ public RelNode accept(RexShuttle shuttle) {
     if (this.exps == exps) {
       return this;
     }
+    final RelDataType rowType =
+        RexUtil.createStructType(
```

Comment:
I'm curious about when we need to invoke this `Project#accept(RexShuttle)`, usually people would customize the shuttle logic through extending a new shuttle implementation. And if want to change the project, we would implement a new `visit(Project)`.

Reply:
In the real-life scenario when I found this issue, I had a RelHomogeneousShuttle with a RexShuttle performing some modifications, and I saw that the Project copies maintain the "old" rowType, even though they have been modified by the shuttle (as the unit test example):
```
class MyRelNodeShuttle extends RelHomogeneousShuttle {
	private final RexShuttle myRexShuttle= new RexShuttle() {
		...
	};

	@Override public final RelNode visit(RelNode other) {
		RelNode next = super.visit(other);
		return next.accept(this.myRexShuttle);
	}
}
```
`RelNode relNode2 = relNode.accept(new MyRelNodeShuttle());`

Reply:
So you mean the case where we visit and update only the project's project list but not the whole project, that make sense.

Reply:
@danny0405 , yes, the Project itself is not modified, its list of expressions is